lib/Twig/Extension/Core.php

```diff
@@ -1507,12 +1507,15 @@ function twig_get_attribute(Twig_Environment $env, Twig_Source $source, $object,
                 continue;
             }
 
-            if (!isset($cache[$name])) {
-                $cache[$name] = $method;
-                $cache[$lcName] = $method;
+            if (!isset($classCache[$name])) {
+                $classCache[$name] = $method;
+            }
+
+            if (!isset($classCache[$lcName])) {
```

Comment:
@fabpot Actually, using a local `$classCache` variable does not fix the issue reported in #2343. It only prevents wrong "global" cache entries. The actual fix is the additional `isset()` call for the lowercased name.